service/datastore/internal/src/main/java/org/eclipse/kapua/service/datastore/internal/MessageStoreServiceImpl.java

```diff
@@ -166,6 +167,7 @@ public DatastoreMessage find(KapuaId scopeId, StorableId id, StorableFetchStyle
     public MessageListResult query(MessageQuery query)
             throws KapuaException {
         checkDataAccess(query.getScopeId(), Actions.read);
+        ArgumentValidator.numRange(query.getLimit(), 0, MAX_LIMIT_VALUE, "limit");
```

Comment:
Does the limit change from being nullable to be mandatory?

Reply:
A request to `.../v1/_/data/messages?sortDir=DESC&offset=0` (so without the `limit` parameter) would still be ok, since the `limit` parameter has a default value of 50.